atomic_reactor/source.py

```diff
@@ -31,8 +32,29 @@
 VcsInfo = collections.namedtuple('VcsInfo', ['vcs_type', 'vcs_url', 'vcs_ref'])
 
 
+class SourceConfig(object):
+    """ read container.yaml file from build source and store as attrs """
+
+    def __init__(self, build_path):
+        self.data = {}
+        file_path = os.path.join(build_path, REPO_CONTAINER_CONFIG)
+        if os.path.exists(file_path):
+            try:
+                # read file and validate against schema
+                self.data = read_yaml_from_file_path(file_path, 'schemas/container.json') or {}
+            except Exception:
+                logger.exception("Failed to load and validate source config YAML from " + file_path)
+                # and proceed with default source config, that is, none
```

Comment:
It would make sense to fail the build at this point (or is there another good alternative?). I am just thinking of those users who might already have a config that doesn't validate, but works given the existing lack of validation. They'll be forced to understand the schema and fix their config to do any kind of build. If that's an acceptable UX... sure, just propagate the exception. Now does seem like the least risky time to start enforcing validation.